### ISSUE TITLE
Can see all information in my profile if the current user isn't authorized (automatically logged out)

### ISSUE THREAD

**Issue by aleksandrakorolova** [scandipwa/scandipwa#1794]
**Description**:
Can see all information in my profile if the current user isn't authorized 

**Expected behavior**:
Home page opens with the notification message 

**To Reproduce**:
1. Log in 
2. Wait some time when you will log out automatically 
3. Open profile
4. Open orders list/wishlist/address book

**Actual result:**:
The loader is displayed Dashboard page + can't see user's information
For other pages can see all information - orders, addresses, products in the wishlist
https://drive.google.com/file/d/1gtW0bZri-X5Bz0vslbRjYMCTibheI8N2/view?usp=sharing

**Note**
Error message 'Internal server error' appears when adding product to cart If he didn't tap a logout

**Comment by zans-laksa** [comment]
In magento every customer has it's own active cart token, that is returned by the `createEmptyCart` mutation.
If customer is not authorized, it is unique on each request, while for authorized customer it is the same.
In magento a unique cart token is created if none is provided, but after login customer cart token is fetched and they both (guest + customer tokens) are merged (in case if guest cart had some items). In this case we have auth and cart tokens created almost at the same time.
While on our current implementation, a new cart token is created and merged to customer after login, without fetching the actual customer cart token. Also tokens are created at the different time. This might lead to no such cart issue.
The other issue is the mix of use redux isSignedIn value and Util/Auth/isSignedIn function calls. While the former is static and updated only when the corresponding action is dispatched, the later is fetching the value from the browser database (localStorage), that have some record expiration checks. And if token is expired, then this function dispatches the logout action. Thus some components do logout on token expiration, while other components rely on a value without any check whether it is expired.
Another issue is on logout. When customer is logged out automatically, there are multiple same requests being sent at the same time.
One more issue is with the server exception parsing. First one is that we cannot rely on debugMessage property, since it isn't available on the production mode. The second one is that some server messages are localized and we cannot parse them.
I've found this code at the cart dispatcher, that was checking whether backend tells us that there is no such cart and tries to create a new one. I've tried to investigate, how magento pwa theme is handling this. The are just sending requests like `query ($cartId:String!){cart(cart_id:$cartId){id __typename}}`. If cart exists and customer is allowed to use it, we get id back. If no, we just get an error (and id is null). But this request is sent only at the page load. Then, after customer session expires, they just ignore any errors.
